kornia/geometry/transform/thin_plate_spline.py

```diff
@@ -0,0 +1,148 @@
+    of (x, y) coordinates. The kernel weights are a (B, K, 2) tensor, and the affine weights are a (B, 3, 2) tensor.
+    For the weight tensors, tensor[..., 0] contains the weights for the x-transform and tensor[..., 1] the weights
+    for the y-transform.
+
+    Returns a (B, N, 2) tensor containing the warped source points.
+
+    Args:
+        points_src (torch.Tensor): tensor of source points (B, N, 2)
+        kernel_centers (torch.Tensor): tensor of kernel center points (B, K, 2)
+        kernel_weights (torch.Tensor): tensor of kernl weights (B, K, 2)
+        affine_weights (torch.Tensor): tensor of affine weights (B, 3, 2)
+
+    Returns:
+        torch.Tensor: The warped source points, from applying the TPS transform.
+    """
+    # f_{x|y}(v) = a_0 + [a_x a_y].v + \sum_i w_i * U(||v-u_i||)
+    pair_distance: torch.Tensor = _pair_square_euclidean(points_src, kernel_centers)
+    k_matrix: torch.Tensor = _kernel_distance(pair_distance)
+
+    # broadcast the kernel distance matrix against the x and y weights to compute the x and y
+    # transforms simultaneously
+    warped: torch.Tensor = (
+        k_matrix[..., None].mul(kernel_weights[:, None]).sum(-2) +
+        points_src[..., None].mul(affine_weights[:, None, 1:]).sum(-2) +
+        affine_weights[:, None, 0]
+    )
+
+    return warped
+
+
+def warp_img_tensor_tps(tensor: torch.Tensor, kernel_centers: torch.Tensor, kernel_weights: torch.Tensor,
+                        affine_weights: torch.Tensor, align_corners: bool = False) -> torch.Tensor:
+    r"""Warp an image tensor according to the thin plate spline transform defined by kernel centers,
+    kernel weights, and affine weights.
+
+    The transform is applied to each pixel coordinate in the output image to obtain a point in the input
+    image for interpolation of the output pixel. So the TPS parameters should correspond to a warp from
+    output space to input space.
+
+    The input tensor is a (B, C, H, W) tensor. The kernel centers, kernel weight and affine weights are the
+    same as in ``warp_points_tps``.
+
+    Returns a warped image tensor with the same shape as the input.
+
+    Args:
+        tensor (torch.Tensor): input image tensor (B, C, H, W)
+        kernel_centers (torch.Tensor): kernel center points (B, K, 2)
+        kernel_weights (torch.Tensor): tensor of kernl weights (B, K, 2)
+        affine_weights (torch.Tensor): tensor of affine weights (B, 3, 2)
+
+    Returns:
+        torch.Tensor: warped image tensor (B, C, H, W)
+    """
+    device, dtype = _extract_device_dtype([tensor, kernel_centers, kernel_weights, affine_weights])
+    batch_size, _, h, w = tensor.shape
+    ys, xs = torch.meshgrid(torch.linspace(-1, 1, h, device=device, dtype=dtype),
+                            torch.linspace(-1, 1, w, device=device, dtype=dtype))
+    coords: torch.Tensor = torch.stack([xs, ys], -1).view(-1, 2)
+    coords = torch.stack([coords] * batch_size, 0)  # expand to batch dimension
```

Comment:
we have [kornia.utils.create_meshrid](https://kornia.readthedocs.io/en/latest/utils.html#kornia.utils.create_meshgrid)